### PR TITLE
pom.xml / JAR

### PR DESCRIPTION
**Pull request by zvenon**
Möchte mit diesem PR Issue #8 abarbeiten.
Konnte erfolgreich eine JAR bilden.

### CODE REVIEW

**Comment by zvenon** [comment]
Nach Eingabe "mvn test" erscheint ständig "no test to run"

**Comment by zvenon** [comment]
screenshot 3
Sorry ,schlechte Quali. 
Richtiger Weg?

**Comment by nipafx** [comment]
Die Ordner `.idea` und `target` sollen nicht commitet werden. Um sie zu entfernen, schließt du am besten IntelliJ, löschst die beiden Ordner (kann sein, dass du das Projekt deswegen später neu importieren musst) und machst dafür einen neuen Commit. Alternativ kannst du diesen PR schließen und einen neuen anlegen - dann entweder die Ordner nicht commiten oder auf #12 warten.

Zu den Tests: Der Pfad in der POM für `testSourceDirectory` ist falsch.

**Comment by FlixSR** [comment]
Ich bekomme bei dem  mvn test
woran könnte das liegen ?  `NFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 6.188 s
[INFO] Finished at: 2017-10-19T10:22:58+02:00
[INFO] Final Memory: 23M/76M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.1.2:test-compile (test-compile) on project THBingen: Compilation failure
[ERROR] java.lang.reflect.InaccessibleObjectException: Unable to make field private static final jdk.internal.misc.Unsafe jdk.internal.misc.Unsafe.theUnsafe accessible: module java.base does not "opens jdk.internal.misc" to unnamed module @12fe1f28
[ERROR]         at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:337)
[ERROR]         at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:281)
[ERROR]         at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:176)
[ERROR]         at java.base/java.lang.reflect.Field.setAccessible(Field.java:170)
[ERROR]         at com.intellij.util.ref.DebugReflectionUtil.getAllFields(DebugReflectionUtil.java:67)
[ERROR]         at com.intellij.util.ref.DebugReflectionUtil.queueStronglyReferencedValues(DebugReflectionUtil.java:147)
[ERROR]         at com.intellij.util.ref.DebugReflectionUtil.walkObjects(DebugReflectionUtil.java:137)
[ERROR]         at com.intellij.util.CachedValueLeakChecker.findReferencedPsi(CachedValueLeakChecker.java:69)
[ERROR]         at com.intellij.util.CachedValueLeakChecker.checkProvider(CachedValueLeakChecker.java:51)
[ERROR]         at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:57)
[ERROR]         at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:140)
[ERROR]         at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:127)
[ERROR]         at com.intellij.psi.impl.source.ClassInnerStuffCache.getConstructors(ClassInnerStuffCache.java:53)
[ERROR]         at com.intellij.psi.impl.compiled.ClsClassImpl.getConstructors(ClsClassImpl.java:179)
[ERROR]         at org.jetbrains.kotlin.load.java.structure.impl.JavaClassImpl.getConstructors(JavaClassImpl.kt:90)
[ERROR]         at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaClassMemberScope$constructors$1.invoke(LazyJavaClassMemberScope.kt:80)
[ERROR]         at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaClassMemberScope$constructors$1.invoke(LazyJavaClassMemberScope.kt:63)
[ERROR]         at org.jetbrains.kotlin.storage.LockBasedStorageManager$LockBasedLazyValue.invoke(LockBasedStorageManager.java:323)
[ERROR]         at org.jetbrains.kotlin.storage.LockBasedStorageManager$LockBasedNotNullLazyValue.invoke(LockBasedStorageManager.java:364)
[ERROR]         at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaClassDescriptor.getConstructors(LazyJavaClassDescriptor.kt:120)
[ERROR]         at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaClassDescriptor.getConstructors(LazyJavaClassDescriptor.kt:53)
[ERROR]         at org.jetbrains.kotlin.resolve.calls.CallResolver.resolveCallForConstructor(CallResolver.java:352)
[ERROR]         at org.jetbrains.kotlin.resolve.calls.CallResolver.resolveFunctionCall(CallResolver.java:290)
[ERROR]         at org.jetbrains.kotlin.resolve.calls.CallResolver.resolveFunctionCall(CallResolver.java:261)
[ERROR]         at org.jetbrains.kotlin.resolve.AnnotationResolverImpl.resolveAnnotationCall(AnnotationResolverImpl.java:151)
[ERROR]         at org.jetbrains.kotlin.resolve.lazy.descriptors.LazyAnnotationDescriptor.computeValueArguments(LazyAnnotations.kt:143)
[ERROR]         at org.jetbrains.kotlin.resolve.lazy.descriptors.LazyAnnotationDescriptor.access$computeValueArguments(LazyAnnotations.kt:108)
[ERROR]         at org.jetbrains.kotlin.resolve.lazy.descriptors.LazyAnnotationDescriptor$valueArguments$1.invoke(LazyAnnotations.kt:126)
[ERROR]         at org.jetbrains.kotlin.resolve.lazy.descriptors.LazyAnnotationDescriptor$valueArguments$1.invoke(LazyAnnotations.kt:108)
[ERROR]         at org.jetbrains.kotlin.storage.LockBasedStorageManager$LockBasedLazyValue.invoke(LockBasedStorageManager.java:323)
[ERROR]         at org.jetbrains.kotlin.storage.LockBasedStorageManager$LockBasedNotNullLazyValue.invoke(LockBasedStorageManager.java:364)
[ERROR]         at org.jetbrains.kotlin.resolve.lazy.descriptors.LazyAnnotationDescriptor.getAllValueArguments(LazyAnnotations.kt:140)
[ERROR]         at org.jetbrains.kotlin.resolve.lazy.descriptors.LazyAnnotationDescriptor.forceResolveAllContents(LazyAnnotations.kt:161)
[ERROR]         at org.jetbrains.kotlin.resolve.lazy.ForceResolveUtil.doForceResolveAllContents(ForceResolveUtil.java:75)
[ERROR]         at org.jetbrains.kotlin.resolve.lazy.ForceResolveUtil.forceResolveAllContents(ForceResolveUtil.java:68)
[ERROR]         at org.jetbrains.kotlin.resolve.LazyTopDownAnalyzer.createFunctionDescriptors(LazyTopDownAnalyzer.kt:254)
[ERROR]         at org.jetbrains.kotlin.resolve.LazyTopDownAnalyzer.analyzeDeclarations(LazyTopDownAnalyzer.kt:190)
[ERROR]         at org.jetbrains.kotlin.resolve.LazyTopDownAnalyzer.analyzeDeclarations$default(LazyTopDownAnalyzer.kt:56)
[ERROR]         at org.jetbrains.kotlin.resolve.jvm.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:104)
[ERROR]         at org.jetbrains.kotlin.resolve.jvm.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:76)
[ERROR]         at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.analyze(KotlinToJVMBytecodeCompiler.kt:347)
[ERROR]         at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:104)
[ERROR]         at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:336)
[ERROR]         at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyzeAndGenerate(KotlinToJVMBytecodeCompiler.kt:322)
[ERROR]         at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileBunchOfSources(KotlinToJVMBytecodeCompiler.kt:208)
[ERROR]         at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:198)
[ERROR]         at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:54)
[ERROR]         at org.jetbrains.kotlin.cli.common.CLICompiler.exec(CLICompiler.java:178)
[ERROR]         at org.jetbrains.kotlin.maven.KotlinCompileMojoBase.execCompiler(KotlinCompileMojoBase.java:220)
[ERROR]         at org.jetbrains.kotlin.maven.K2JVMCompileMojo.execCompiler(K2JVMCompileMojo.java:180)
[ERROR]         at org.jetbrains.kotlin.maven.K2JVMCompileMojo.execCompiler(K2JVMCompileMojo.java:43)
[ERROR]         at org.jetbrains.kotlin.maven.KotlinCompileMojoBase.execute(KotlinCompileMojoBase.java:203)
[ERROR]         at org.jetbrains.kotlin.maven.KotlinTestCompileMojo.execute(KotlinTestCompileMojo.java:87)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
[ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
[ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
[ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
[ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:564)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[ERROR] 
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
`

**Comment by zvenon** [comment]
Fixes #12  ?

**Comment by zvenon** [comment]
Habe zusätzlich im Ordner 'test' einen Ordner namens 'kotlin' erstellt und diesen als test source root markiert, jenes auch im Pfad in der pom abgeändert. Jedoch funktioniert dann der Test nicht mehr. 
--> Maven: "only the Kotlin standard libary is allowed to use 'kotlin' package"

**Comment by zvenon** [comment]
wenn ich alles lasse:
<testSourceDirectory>${project.basedir}/src/test</testSourceDirectory>
funktioniert der Test 😁

**Comment by zvenon** [comment]
@nicolaiparlog  Hast du einen Tipp wie ich die POM bzw `testSourceDirectory` ändern soll?

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>de.thbingen.info2</groupId>
+    <artifactId>THBingen</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <kotlin.version>1.1.2</kotlin.version>
```
Comment (nipafx) [inline pom.xml]: @FlixsTHB: Du hast anscheinend Java 9 installiert. Kann sein, dass sich das Problem durch eine aktuellere Kotlin-Version (hier `1.1.51` eintragen) lösen lässt. Sonst installier bitte [JDK 8](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html).

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <sourceDirectory>${project.basedir}/src/main/kotlin</sourceDirectory>
+        <testSourceDirectory>${project.basedir}/src/test</testSourceDirectory>
```
Comment (nipafx) [inline pom.xml]: @zvenon: 

> Habe zusätzlich im Ordner 'test' einen Ordner namens 'kotlin' erstellt

Gut.

> und diesen als test source root markiert, jenes auch im Pfad in der pom abgeändert.

Wenn das Projekt als Maven-Projekt importiert ist, liest IntelliJ dinge wie Test Source Roots aus der POM aus und überschreibt damit andere Einstellungen - du solltest das also nur in der POM ändern.

> Jedoch funktioniert dann der Test nicht mehr.
> --> Maven: "only the Kotlin standard libary is allowed to use 'kotlin' package"

Das klingt so als würde der Ordner `kotlin` dann als Paket interpretiert. Das dürfen wir aber nicht benutzen, sondern nur Kotlin selber.

Es müsste gehen, wenn du nochmal den Ordner `kotlin` anlegst, `HelloWorldTest` dorthinein verschiebst (sicherstellen, dass die Datei nicht mit `package kotlin` beginnt - wenn doch die Zeile löschen) und dann den `testSourceDirectory` Pfad ebenfalls um `kotlin` erweiterst.

File: pom.xml (hunk 1 above)
Comment (nipafx) [inline pom.xml]: Unabhängig von @FlixsTHB's Problem, wäre es vielleicht ganz gut, hier auf `1.1.51` zu aktualisieren.

**Review by nipafx — CHANGES_REQUESTED** [review]
Sieht gut aus, inklusive ausführbarer JAR. Zwei Kleinigkeiten noch:

File: pom.xml (hunk 1 above)
Comment (FlixSR) [reply pom.xml]: ja hab ich gelöst lag an der kotlin Version Danke

**Comment by FlixSR** [comment]
Kotlinversionsupdate wurde durchgeführt

**Comment by FlixSR** [comment]
@nicolaiparlog der Test wurde verschoben und sollte funktionieren, könnte somit freigegeben werden.

**Comment by zvenon** [comment]
Super, so funktioniert der Test bei mir auch! 
-ohne "package" Fehrleranzeige! 👍 
Kann dieser PR dann gemerged werden?

**Review by nipafx — APPROVED** [review]

**Comment by nipafx** [comment]
:+1:

**Pull request merged**